File: 1-yahtzee-java/src/main/java/yahtzee/DiceRoller.java
package yahtzee;

import java.util.LinkedHashMap;
import java.util.Map;

public class DiceRoller {

    private DieRoller dieRoller;
    private Map<Dice, Integer> rollResult;

    public DiceRoller(DieRoller dieRoller) {
        this.rollResult = new LinkedHashMap<Dice, Integer>();
        this.dieRoller = dieRoller;
    }

    public void roll(Dice... dice) {
        for (Dice die : dice) {
            rollResult.put(die, this.dieRoller.roll());
        }
    }

    public Map<Dice, Integer> lastRolledDice() {
        return this.rollResult;
    }
}

File: 2-yahtzee-php/src/Yahtzee/DiceRoller.php
<?php


namespace Yahtzee;


class DiceRoller
{
    /** @var array */
    private $rollResult;
    
    /** @var DieRoller */
    private $dieRoller;

    /**
     * @param DieRoller $dieRoller
     */
    public function __construct(DieRoller $dieRoller)
    {
        $this->dieRoller = $dieRoller;
        $this->rollResult = [];
    }

    /**
     * @return array
     */
    public function lastRollResult()
    {
        return $this->rollResult;
    }

    public function rollAll()
    {
        $this->rollDice([
            Dice::D1(),
            Dice::D2(),
            Dice::D3(),
            Dice::D4(),
            Dice::D5()
        ]);
    }

    /**
     * @param Dice[] $diceToReRun
     */
    public function reRun($diceToReRun)
    {
        $this->rollDice($diceToReRun);
    }

    /**
     * @param Dice[] $diceToRoll
     */
    private function rollDice($diceToRoll)
    {
        foreach ($diceToRoll as $die) {
            $this->rollResult[(string) $die] = $this->dieRoller->roll();
        }
    }
}
